benchopt/util.py

```diff
@@ -103,42 +92,50 @@ def _run_bash_in_env(script, env_name=None, raise_on_error=None,
                         capture_stdout=capture_stdout)
 
 
-def pip_install_in_env(*packages, env_name=None):
-    """Install the packages with pip in the given environment"""
-    if env_name is None and not ALLOW_INSTALL:
-        raise ValueError("Trying to install solver not in a virtualenv. "
-                         "To allow this, set BENCHO_ALLOW_INSTALL=True.")
-    cmd = PIP_INSTALL_CMD.format(packages=' '.join(packages))
-    error_msg = f"Failed to pip install packages {packages}\nError:{{output}}"
-    _run_bash_in_env(cmd, env_name=env_name,
-                     raise_on_error=error_msg)
-
-
-def pip_uninstall_in_env(*packages, env_name=None):
-    """Uninstall the packages with pip in the given environment"""
+# TODO: suggestion new name: install_in_conda_env (since it also uses pip)
+def conda_install_in_env(*packages, env_name=None, force=False):
+    """Install the packages with conda in the given environment"""
     if env_name is None and not ALLOW_INSTALL:
-        raise ValueError("Trying to uninstall solver not in a virtualenv. "
+        raise ValueError("Trying to install solver not in a conda env. "
                          "To allow this, set BENCHO_ALLOW_INSTALL=True.")
-    cmd = PIP_UNINSTALL_CMD.format(packages=' '.join(packages))
-    _run_bash_in_env(cmd, env_name=env_name,
-                     raise_on_error=f"Failed to uninstall packages {packages}"
-                     "\nError: {output}")
+    install_this = False
+    if '-e .' in packages:
+        install_this = True
+        packages = list(packages)
+        packages.remove('-e .')
+
+    pip_packages = [pkg[4:] for pkg in packages if pkg.startswith('pip:')]
+    conda_packages = [pkg for pkg in packages if not pkg.startswith('pip:')]
+
+    error_msg = (f"Failed to conda install packages {packages}\n"
+                 "Error:{output}")
+    if conda_packages:
+        cmd = CONDA_INSTALL_CMD.format(packages=' '.join(conda_packages))
+        if force:
+            cmd += ' --force-reinstall'
+        _run_bash_in_env(cmd, env_name=env_name, raise_on_error=error_msg)
+    if pip_packages:
+        cmd = PIP_INSTALL_CMD.format(packages=' '.join(pip_packages))
+        if force:
+            cmd += ' --force-reinstall'
+        _run_bash_in_env(cmd, env_name=env_name, raise_on_error=error_msg)
+    if install_this:
+        _run_bash_in_env('pip install -e .', env_name=env_name)
 
 
 def bash_install_in_env(script, env_name=None):
     """Run a bash install script in the given environment"""
     if env_name is None and not ALLOW_INSTALL:
-        raise ValueError("Trying to install solver not in a virtualenv. "
+        raise ValueError("Trying to install solver not in a conda env. "
                          "To allow this, set BENCHO_ALLOW_INSTALL=True.")
-    env = "$VIRTUAL_ENV" if env_name is not None else "$HOME/.local/"
+    env = env_name if env_name is not None else "base"
+    # TODO correct idea to use base?
```

Comment:
@tomMoral I am not 100 % sure here

Reply:
Can't we use the current `env` in this case?